### PR TITLE
fix: ignore casing when checking bt-expose-raw-proxy-response header

### PR DESCRIPTION
**Pull request by lcschy**
## Description

- Title +
- Update simulator and xcode versions
- Fix issue w/ checking for invalid URL would not throw
- Update tests to match new versions

## Testing required outside of automated testing?

- [x] Not Applicable

### Screenshots (if appropriate):

- [x] Not Applicable

## Rollback / Rollforward Procedure

- [x] Roll Forward
- [ ] Roll Back

## Reviewer Checklist

- [ ] Description of Change
- [ ] Description of outside testing if applicable.
- [ ] Description of Roll Forward / Backward Procedure
- [ ] Documentation updated for Change

### CODE REVIEW

File: BasisTheoryElements/Sources/BasisTheoryElements/HttpClientHelpers.swift [hunk 1]
```diff
@@ -73,12 +73,16 @@ private func encodeParamsToFormUrlEncoded(_ formParams: [String: Any]) -> String
 
 struct HttpClientHelpers {
     static func executeRequest(method: HttpMethod, url: String, payload: [String: Any]?, config: Config?, completion: @escaping ((_ request: URLResponse?, _ data: JSON?, _ error: Error?) -> Void)) -> Void {
-        guard let url = URL(string: url) else {
+        guard var components = URLComponents(string: url), components.scheme != nil else {
```
Comment (lcschy) [inline BasisTheoryElements/Sources/BasisTheoryElements/HttpClientHelpers.swift:76]: I guess this changed w/ OS? But it was considering that crazy string a valid URL, had to make it more complex.

File: IntegrationTester/AcceptanceTests/TextElementUITextFieldTests.swift [hunk 2]
```diff
@@ -79,13 +79,11 @@ final class IntegrationTesterUITests: XCTestCase {
         
         nameFieldCopyIcon.tap()
         
-        // doubleTap doesn't work because it has a certain delay to display the menu
-        phoneNumberTextField.tap()
-        phoneNumberTextField.tap()
+        phoneNumberTextField.press(forDuration: 1)
```
Comment (lcschy) [inline IntegrationTester/AcceptanceTests/TextElementUITextFieldTests.swift:82]: This seemed more reliable than the double tap on new simulator.

File: IntegrationTester/UnitTests/ProxyServiceTests.swift [hunk 3]
```diff
@@ -448,7 +448,7 @@ final class ProxyServiceTests: XCTestCase {
             do {
                 let token = try result.get().body.data!.value as! [String: Any]
                 
-                XCTAssertEqual(token["textFieldRef"] as! String, "https://echo.basistheory.com/get")
+                XCTAssertEqual(token["textFieldRef"] as! String, "http://echo.basistheory.com/get")
```
Comment (lcschy) [inline IntegrationTester/UnitTests/ProxyServiceTests.swift:451]: This was a change directly on the echo server, so had to change here to match.

File: .github/scripts/integration-test.sh [hunk 4]
```diff
@@ -27,5 +27,5 @@ xcodebuild clean test \
     -project ./IntegrationTester/IntegrationTester.xcodeproj \
     -scheme IntegrationTester \
     -configuration Debug \
-    -destination platform="iOS Simulator,OS=16.1,name=iPhone 14 Pro" \
+    -destination platform="iOS Simulator,OS=latest,name=iPhone 15 Pro" \
```
Comment (kevinperaza) [inline .github/scripts/integration-test.sh:30]: This is fine, but shouldn't we be testing against the lowest version we support?

**Review by kevinperaza — DISMISSED** [review]

File: .github/scripts/integration-test.sh (hunk 4 above)
Comment (lcschy) [reply .github/scripts/integration-test.sh]: I went as low as possible w/ the machine/xcode version combo rn.
We can look into what it takes to have 'lesser' iOS versions later on

**Review by kevinperaza — APPROVED** [review]

**Pull request merged**